### PR TITLE
Add Router to SwiftKit.

### PR DESCRIPTION
**Pull request by TadeasKriz**

### CODE REVIEW

File: Router/Router.swift [hunk 1]
```diff
@@ -0,0 +1,539 @@
...
+public protocol MappableRouteTarget {
+    typealias Model: Mappable
+}
+
+public protocol RouteHeaders {
+    var headers: [String: AnyObject] { get }
+}
+
+/// Protocol to define the opaque type returned from a request
+public protocol Cancellable {
```
Comment (FilipDolnik) [inline Router/Router.swift]: Why not delete this, rename `CancellableToken` to `Cancellable`,make it public and cancelAction private?

File: Router/Router.swift [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+    public func request(target: T, completion: RouterCompletion) -> Cancellable {
+        var headers: [String: AnyObject] = [:]
+        
+        var url = target.baseURL.URLByAppendingPathComponent(target.endpoint.path())
+
+        var request: NSMutableURLRequest = NSMutableURLRequest(URL: url)
+        request.HTTPMethod = target.endpoint.method().rawValue
+        
+        request.allHTTPHeaderFields = headers
```
Comment (FilipDolnik) [inline Router/Router.swift]: Why not `request.allHTTPHeaderFields = [:]` and remove `var headers: [String: AnyObject] = [:]` ?

File: Router/Router.swift [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+public typealias RouterCompletion = (data: NSData?, statusCode: Int?, response: NSURLResponse?, error: NSError?) -> ()
+
+/// Protocol to define the base URL and sample data for an enum.
+public protocol RouteTarget {
+    var baseURL: NSURL { get }
+    var endpoint: Route.Endpoint { get }
+    var parameters: Route.Parameters { get }
+}
+
+public protocol MappableRouteTarget {
```
Comment (FilipDolnik) [inline Router/Router.swift]: Where is this used?

File: Router/Router.swift [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+        self.statusCode = statusCode
+        self.response = response
+        self.error = error
+        self.rawData = rawData
+    }
+    
+}
+
+/// Block to be executed when a request has completed.
+public typealias RouterCompletion = (data: NSData?, statusCode: Int?, response: NSURLResponse?, error: NSError?) -> ()
```
Comment (FilipDolnik) [inline Router/Router.swift]: I would move this to Router.

File: Router/Router.swift (hunk 2 above)
Comment (TadeasKriz) [inline Router/Router.swift]: Because we want to add support for RouteHeaders protocol before releasing this.

File: Router/Endpoint.swift [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+//
+
+import Foundation
+import Alamofire
+
+public protocol Endpoint {
+    typealias Input
+    typealias Output
+    
+    init(_ path: String)
```
Comment (FilipDolnik) [inline Router/Endpoint.swift:16]: I would remove this initializer. If you want to use it, create new class. That way BaseEndpoint doesn't have to implement it.

Comment (TadeasKriz) [inline Router/Endpoint.swift:16]: It is there because it is used by Target class.

**Pull request merged**